### PR TITLE
Test/complete test suite

### PR DESCRIPTION
**Pull request by Roman-batut**
This pull request includes several updates to the test files for various components, focusing on improving the test structure, adding new utilities, and ensuring consistency across the tests. The most important changes include renaming test files, updating imports, adding new mock implementations, and enhancing the test cases to use `screen` from `@testing-library/react-native`.

### Test File Renaming and Import Updates:
* [`__tests__/Components/home/challenge.test.tsx`](diffhunk://#diff-3744e1b8977aa69120016187bc9c7e47a545fd4af5cf5f892a4d5bfc658ae064L2-R8): Renamed from `__tests__/View-tests/challenge.test.tsx` and updated imports to include `screen` and `act` from `@testing-library/react-native`.
* [`__tests__/View-tests/auth/setUsername.test.tsx`](diffhunk://#diff-9222ce0b7bbe8fbd43372b72ddd438715d7429c10c9ed82a78b2edf41090cce4L2-R7): Updated imports to include `screen` and `waitFor` from `@testing-library/react-native`.
* [`__tests__/View-tests/home/homeScreen.test.tsx`](diffhunk://#diff-e1f2127af597fcae0867a4626a056c41e508b23e0735b49062f677d557a2d5c7L2-R7): Updated imports to include `screen` and `waitFor` from `@testing-library/react-native`.

### Mock Implementations:
* [`__tests__/View-tests/home/homeScreen.test.tsx`](diffhunk://#diff-e1f2127af597fcae0867a4626a056c41e508b23e0735b49062f677d557a2d5c7R34-R39): Added a new mock implementation for the `getUser` method in `FirestoreCtrl`.
* [`__tests__/View-tests/welcome/finalScreen.test.tsx`](diffhunk://#diff-7e2ad7558c2ad2848def033a5c05cc636ac9dffe60831b6f029a79da06bdc9a6R5-R20): Added a mock implementation for `FirestoreCtrl` and used it in multiple test cases. [[1]](diffhunk://#diff-7e2ad7558c2ad2848def033a5c05cc636ac9dffe60831b6f029a79da06bdc9a6R5-R20) [[2]](diffhunk://#diff-7e2ad7558c2ad2848def033a5c05cc636ac9dffe60831b6f029a79da06bdc9a6L18-R34) [[3]](diffhunk://#diff-7e2ad7558c2ad2848def033a5c05cc636ac9dffe60831b6f029a79da06bdc9a6L31-R47)

### Enhancements to Test Cases:
* [`__tests__/Components/home/challenge.test.tsx`](diffhunk://#diff-3744e1b8977aa69120016187bc9c7e47a545fd4af5cf5f892a4d5bfc658ae064L102-R110): Enhanced test cases to use `screen` for querying elements and added `waitFor` and `act` for handling asynchronous events. [[1]](diffhunk://#diff-3744e1b8977aa69120016187bc9c7e47a545fd4af5cf5f892a4d5bfc658ae064L102-R110) [[2]](diffhunk://#diff-3744e1b8977aa69120016187bc9c7e47a545fd4af5cf5f892a4d5bfc658ae064L114-R136) [[3]](diffhunk://#diff-3744e1b8977aa69120016187bc9c7e47a545fd4af5cf5f892a4d5bfc658ae064L139-R157)
* [`__tests__/View-tests/auth/setUsername.test.tsx`](diffhunk://#diff-9222ce0b7bbe8fbd43372b72ddd438715d7429c10c9ed82a78b2edf41090cce4R21-R34): Refactored test cases to use a `SetUsernameTest` component for rendering and used `screen` for querying elements. [[1]](diffhunk://#diff-9222ce0b7bbe8fbd43372b72ddd438715d7429c10c9ed82a78b2edf41090cce4R21-R34) [[2]](diffhunk://#diff-9222ce0b7bbe8fbd43372b72ddd438715d7429c10c9ed82a78b2edf41090cce4L31-R60) [[3]](diffhunk://#diff-9222ce0b7bbe8fbd43372b72ddd438715d7429c10c9ed82a78b2edf41090cce4L68-R77)
* [`__tests__/View-tests/home/homeScreen.test.tsx`](diffhunk://#diff-e1f2127af597fcae0867a4626a056c41e508b23e0735b49062f677d557a2d5c7L70-R82): Updated test cases to use `waitFor` for handling asynchronous rendering. [[1]](diffhunk://#diff-e1f2127af597fcae0867a4626a056c41e508b23e0735b49062f677d557a2d5c7L70-R82) [[2]](diffhunk://#diff-e1f2127af597fcae0867a4626a056c41e508b23e0735b49062f677d557a2d5c7R97)

These changes improve the readability, maintainability, and reliability of the test cases by ensuring consistent usage of testing utilities and better handling of asynchronous operations.This pull request includes significant updates to multiple test files to improve the consistency and reliability of the tests. The most important changes involve renaming files, importing additional methods from `@testing-library/react-native`, and updating the usage of test utilities.

### Test File Updates:

* [`__tests__/Components/home/challenge.test.tsx`](diffhunk://#diff-3744e1b8977aa69120016187bc9c7e47a545fd4af5cf5f892a4d5bfc658ae064L2-R8): Renamed from `__tests__/View-tests/challenge.test.tsx`, added imports for `screen`, `act`, and updated test cases to use `screen` and `waitFor` for better async handling. [[1]](diffhunk://#diff-3744e1b8977aa69120016187bc9c7e47a545fd4af5cf5f892a4d5bfc658ae064L2-R8) [[2]](diffhunk://#diff-3744e1b8977aa69120016187bc9c7e47a545fd4af5cf5f892a4d5bfc658ae064R71) [[3]](diffhunk://#diff-3744e1b8977aa69120016187bc9c7e47a545fd4af5cf5f892a4d5bfc658ae064L102-R110) [[4]](diffhunk://#diff-3744e1b8977aa69120016187bc9c7e47a545fd4af5cf5f892a4d5bfc658ae064L114-R136) [[5]](diffhunk://#diff-3744e1b8977aa69120016187bc9c7e47a545fd4af5cf5f892a4d5bfc658ae064L139-R157) [[6]](diffhunk://#diff-3744e1b8977aa69120016187bc9c7e47a545fd4af5cf5f892a4d5bfc658ae064L153-R168) [[7]](diffhunk://#diff-3744e1b8977aa69120016187bc9c7e47a545fd4af5cf5f892a4d5bfc658ae064L165-R194)

* [`__tests__/View-tests/auth/setUsername.test.tsx`](diffhunk://#diff-9222ce0b7bbe8fbd43372b72ddd438715d7429c10c9ed82a78b2edf41090cce4L2-R7): Added imports for `screen` and `waitFor`, created a helper component `SetUsernameTest` to simplify rendering, and updated test cases to use `screen` for queries. [[1]](diffhunk://#diff-9222ce0b7bbe8fbd43372b72ddd438715d7429c10c9ed82a78b2edf41090cce4L2-R7) [[2]](diffhunk://#diff-9222ce0b7bbe8fbd43372b72ddd438715d7429c10c9ed82a78b2edf41090cce4R21-R34) [[3]](diffhunk://#diff-9222ce0b7bbe8fbd43372b72ddd438715d7429c10c9ed82a78b2edf41090cce4L31-R60) [[4]](diffhunk://#diff-9222ce0b7bbe8fbd43372b72ddd438715d7429c10c9ed82a78b2edf41090cce4L68-R77) [[5]](diffhunk://#diff-9222ce0b7bbe8fbd43372b72ddd438715d7429c10c9ed82a78b2edf41090cce4L99-R97) [[6]](diffhunk://#diff-9222ce0b7bbe8fbd43372b72ddd438715d7429c10c9ed82a78b2edf41090cce4L130-R115)

* [`__tests__/View-tests/home/homeScreen.test.tsx`](diffhunk://#diff-e1f2127af597fcae0867a4626a056c41e508b23e0735b49062f677d557a2d5c7L2-R7): Added imports for `screen` and `waitFor`, and updated test cases to use `screen` for queries and better async handling. [[1]](diffhunk://#diff-e1f2127af597fcae0867a4626a056c41e508b23e0735b49062f677d557a2d5c7L2-R7) [[2]](diffhunk://#diff-e1f2127af597fcae0867a4626a056c41e508b23e0735b49062f677d557a2d5c7R34-R39) [[3]](diffhunk://#diff-e1f2127af597fcae0867a4626a056c41e508b23e0735b49062f677d557a2d5c7R50) [[4]](diffhunk://#diff-e1f2127af597fcae0867a4626a056c41e508b23e0735b49062f677d557a2d5c7L70-R82) [[5]](diffhunk://#diff-e1f2127af597fcae0867a4626a056c41e508b23e0735b49062f677d557a2d5c7R97) [[6]](diffhunk://#diff-e1f2127af597fcae0867a4626a056c41e508b23e0735b49062f677d557a2d5c7R113)

* [`__tests__/View-tests/welcome/finalScreen.test.tsx`](diffhunk://#diff-7e2ad7558c2ad2848def033a5c05cc636ac9dffe60831b6f029a79da06bdc9a6R5-R20): Mocked `FirestoreCtrl` and updated test cases to use the mocked instance to ensure consistent test results. [[1]](diffhunk://#diff-7e2ad7558c2ad2848def033a5c05cc636ac9dffe60831b6f029a79da06bdc9a6R5-R20) [[2]](diffhunk://#diff-7e2ad7558c2ad2848def033a5c05cc636ac9dffe60831b6f029a79da06bdc9a6L18-R34) [[3]](diffhunk://#diff-7e2ad7558c2ad2848def033a5c05cc636ac9dffe60831b6f029a79da06bdc9a6L31-R47) [[4]](diffhunk://#diff-7e2ad7558c2ad2848def033a5c05cc636ac9dffe60831b6f029a79da06bdc9a6L47-R63) [[5]](diffhunk://#diff-7e2ad7558c2ad2848def033a5c05cc636ac9dffe60831b6f029a79da06bdc9a6L60-R76) [[6]](diffhunk://#diff-7e2ad7558c2ad2848def033a5c05cc636ac9dffe60831b6f029a79da06bdc9a6L73-R89) [[7]](diffhunk://#diff-7e2ad7558c2ad2848def033a5c05cc636ac9dffe60831b6f029a79da06bdc9a6L86-R102) [[8]](diffhunk://#diff-7e2ad7558c2ad2848def033a5c05cc636ac9dffe60831b6f029a79da06bdc9a6L99-R115) [[9]](diffhunk://#diff-7e2ad7558c2ad2848def033a5c05cc636ac9dffe60831b6f029a79da06bdc9a6L114-R130)

### Test File Addition: 

Multiple test files have been added to improve test coverage:
- In ViewModel tests: `forgotPasswordViewModel.test.tsx`, `setUsernameViewModel.test.tsx`, `cameraViewModel.test.tsx`, `maximizeScreenViewModel.test.tsx`, `profileScreenViewModel.test.tsx`, `mapScreenViewModel.test.tsx`, `finalScreenViewModel.test.tsx`, `userStateFlow.test.tsx`
- In types: `auth.test.tsx`, `challengeBuilder.test.tsx`
- In components: `numberCards.test.tsx`, `timer.test.tsx`

### CODE REVIEW

File: __tests__/View-tests/auth/setUsername.test.tsx [hunk 1]
```diff
@@ -96,22 +91,10 @@ describe("SetUsernameScreen Tests", () => {
...
-        navigation={mockNavigation}
-        firestoreCtrl={mockFirestoreCtrl}
-        setUser={mockSetUser}
-      />,
-    );
+    render(<SetUsernameTest />);
 
     // Vérifie que le message d'erreur est affiché
-    const errorMessage = getByText("Failed to pick image.");
+    const errorMessage = screen.getByText("Failed to pick image.");
```
Comment (margdsq) [inline __tests__/View-tests/auth/setUsername.test.tsx:97]: Thank you for finding this !

File: app/src/views/home/home_screen.tsx [hunk 2]
```diff
@@ -93,7 +93,7 @@ export default function HomeScreen({
         {/* Current Challenge Description  */}
         <ChallengeDescription
           dBChallengeDescription={titleChallenge}
-          onTimerFinished={() => console.log("Timer Finished")}
+          onTimerFinished={() => console.info("Timer Finished")}
```
Comment (margdsq) [inline app/src/views/home/home_screen.tsx:96]: What's the difference here ?

File: app/types/ChallengeBuilder.tsx [hunk 3]
```diff
@@ -61,11 +61,10 @@ export const createChallenge = async (
   try {
     // Prepare the challenge data for Firestore
     const user: DBUser = await firestoreCtrl.getUser();
-    console.log("createChallenge uid", user.uid);
-
-    if (location == null) {
-      console.log("location undefined");
-    }
+    //console.log("createChallenge uid", user.uid);
```
Comment (margdsq) [inline app/types/ChallengeBuilder.tsx:64]: Just remove this if it is not necessary !

**Review by margdsq — COMMENTED** [review]
I just have one question but appart from that it's an amazing PR, thank you for this, bringing the test coverage from 40% to 70% surely brings us a lot closer to having a bugless app !!

File: app/src/views/home/home_screen.tsx (hunk 2 above)
Comment (Roman-batut) [reply app/src/views/home/home_screen.tsx:96]: Basically I put it as info as it is not an important comment and I can un-show it safely inside tests.

Comment (margdsq) [reply app/src/views/home/home_screen.tsx:96]: Ok great!!

**Review by margdsq — APPROVED** [review]
Thank you so much for this !!

**Pull request merged**